src/extensions/scratch3_text2speech/index.js

```diff
@@ -376,13 +510,49 @@ class Scratch3Text2SpeechBlocks {
     }
 
     /**
-     * Check if a language code is in the list of supported languages for the
+     * Get the extension locale for a supported locale, or null.
+     * @param {string} locale a locale code.
+     * @returns {?string} a locale supported by the extension.
+     */
+    _getExtensionLocaleForSupportedLocale (locale) {
+        for (const lang in this.LANGUAGE_INFO) {
+            if (this.LANGUAGE_INFO[lang].locales.includes(locale)) {
+                return lang;
+            }
+        }
+        log.error(`cannot find extension locale for locale ${locale}`);
```

Comment:
Should this return some default value or null in the case where it isn't found? Or no because it is only used for saving?  I can't quite tell whether state.textToSpeechLanguage is being set in the same cases as it was before and if not, if that is intentional.

Reply:
The logic for setting `stage.textToSpeechLanguage` is the same as before- the `log.error` should only happen in the case that the `LANGUAGE_INFO` data structure somehow gets out of sync (in which case it's a useful error).